[CI] SearchableSnapshotsIntegTests.testCreateAndRestoreSearchableSnapshot failing

There are a couple of errors for this test in the last week:

https://gradle-enterprise.elastic.co/s/iqg3cqcqalyak
https://gradle-enterprise.elastic.co/s/3ilvp7ty52ybk

Both seems to fail with the same error:

```
 Caused by:
        java.lang.AssertionError: fake allocation id has to be removed, inSyncAllocationIds:[AWIAuDNXRKK3b0jtoJ-ePw, _forced_allocation_]
            at __randomizedtesting.SeedInfo.seed([FBB4D9C024EA5F0A]:0)
            at org.elasticsearch.cluster.routing.allocation.IndexMetadataUpdater.updateInSyncAllocations(IndexMetadataUpdater.java:186)
            at org.elasticsearch.cluster.routing.allocation.IndexMetadataUpdater.applyChanges(IndexMetadataUpdater.java:122)
            at org.elasticsearch.cluster.routing.allocation.RoutingAllocation.updateMetadataWithRoutingChanges(RoutingAllocation.java:238)
            at org.elasticsearch.cluster.routing.allocation.AllocationService.buildResult(AllocationService.java:147)
            at org.elasticsearch.cluster.routing.allocation.AllocationService.buildResultAndLogHealthChange(AllocationService.java:132)
            at org.elasticsearch.cluster.routing.allocation.AllocationService.applyStartedShards(AllocationService.java:128)
            at org.elasticsearch.cluster.action.shard.ShardStateAction$ShardStartedClusterStateTaskExecutor.execute(ShardStateAction.java:579)
            at org.elasticsearch.cluster.service.MasterService.executeTasks(MasterService.java:697)
            at org.elasticsearch.cluster.service.MasterService.calculateTaskOutputs(MasterService.java:319)
            at org.elasticsearch.cluster.service.MasterService.runTasks(MasterService.java:214)
            at org.elasticsearch.cluster.service.MasterService$Batcher.run(MasterService.java:151)
            at org.elasticsearch.cluster.service.TaskBatcher.runIfNotProcessed(TaskBatcher.java:150)
            at org.elasticsearch.cluster.service.TaskBatcher$BatchedTask.run(TaskBatcher.java:188)
            at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:632)
            at org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.runAndClean(PrioritizedEsThreadPoolExecutor.java:252)
            at org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.run(PrioritizedEsThreadPoolExecutor.java:215)
            at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
            at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
            at java.base/java.lang.Thread.run(Thread.java:834)
```

I have not been able to reproduce it:

```
./gradlew ':x-pack:plugin:searchable-snapshots:test' --tests "org.elasticsearch.xpack.searchablesnapshots.SearchableSnapshotsIntegTests.testCreateAndRestoreSearchableSnapshot"   -Dtests.seed=38EA7CA2AD868AF3   -Dtests.security.manager=true   -Dtests.locale=ja-JP-u-ca-japanese-x-lvariant-JP   -Dtests.timezone=Navajo   -Dcompiler.java=14
```

--- a/server/src/main/java/org/elasticsearch/cluster/routing/RecoverySource.java
+++ b/server/src/main/java/org/elasticsearch/cluster/routing/RecoverySource.java
@@ -63,7 +63,7 @@ public abstract class RecoverySource implements Writeable, ToXContentObject {
         Type type = Type.values()[in.readByte()];
         switch (type) {
             case EMPTY_STORE: return EmptyStoreRecoverySource.INSTANCE;
-            case EXISTING_STORE: return new ExistingStoreRecoverySource(in);
+            case EXISTING_STORE: return ExistingStoreRecoverySource.read(in);
             case PEER: return PeerRecoverySource.INSTANCE;
             case SNAPSHOT: return new SnapshotRecoverySource(in);
             case LOCAL_SHARDS: return LocalShardsRecoverySource.INSTANCE;
@@ -152,8 +152,8 @@ public abstract class RecoverySource implements Writeable, ToXContentObject {
             this.bootstrapNewHistoryUUID = bootstrapNewHistoryUUID;
         }
 
-        private ExistingStoreRecoverySource(StreamInput in) throws IOException {
-            bootstrapNewHistoryUUID = in.readBoolean();
+        private static ExistingStoreRecoverySource read(StreamInput in) throws IOException {
+            return in.readBoolean() ? FORCE_STALE_PRIMARY_INSTANCE : INSTANCE;
         }
 
         @Override

